pom.xml

```diff
@@ -98,7 +98,7 @@
         <pmd.plugin.version>3.20.0</pmd.plugin.version>
         <ant.version>1.10.13</ant.version>
         <javadoc.plugin.version>3.4.1</javadoc.plugin.version>
-        <antlr.version>4.7.2</antlr.version>
+        <antlr.version>4.11.1</antlr.version>
```

Comment:
Is this update needed?

Reply:
It is needed indeed as the T-SQL Lexer is case insensitive and this is a feature not present in antlr4 v4.7.2.

Reply:
> but not on the CI where several classes cannot be instantiated.

So, the update of antlr from 4.7.2 to 4.11.1 is a major upgrade. Since [4.10](https://github.com/antlr/antlr4/releases/tag/4.10), antlr requires at least Java 11.

Adding this dependency globally will break other languages, that still assume java7... we test java7 compatibility on the CI only. Upgrading the minimum required java runtime is a major upgrade and can't be done for pmd6.
However, we have some languages that require at least java8 (apex, js, html).

I think, we either need to use a workaround for case insensitive lexer (https://github.com/antlr/antlr4/pull/3399 talks about a case changing char stream) or we need to shade the antlr dependency into pmd-tsql, so that it doesn't leak into the other modules... we could then define, that for pmd-tsql at least java11 is required.

I can give it a try with shading in the antlr dependency...

Reply:
It looks like we need antlr4 >= 4.10 and also that this requires Java 1.8 (and not 11), if I understand this release note properly:
https://github.com/antlr/antlr4/releases/tag/4.10.1

So as a mitigation, could we make tsql require Java 8 and build with antlr4 4.11.1 with java8 and 4.7.2 with java 7?

Reply:
Yes, it seems antlr 4.10 requires only java8 (for build java11). However, the problem is not building, but running. We only have one classpath at runtime, so we can either be compatible with java7 or not.

I've tried out the solution from before antlr 4.10 using a [CaseChangingCharStream](https://github.com/antlr/antlr4/blob/4.9.1/doc/resources/CaseChangingCharStream.java) and this works fine.

I'll update this PR and restore the old antlr version.